design-book/src/design-constraints.md

```diff
@@ -0,0 +1,53 @@
+# Design Constraints
+
+To help ensure that we're building an editor that's both useful to end users and feasible to maintain,
+we've developed a set of design constraints and desiderata. Please try to follow them when making and reviewing changes!
+As we're developing, initial prototypes may bend or break the rules: just be mindful of how we can move towards the final desirable end goal.
+
+Below each bullet point, motivation is given.
+
+## Must
+
+- The editor must target the latest stable Bevy release.
+  - This eases project management by separating the work across two repositories, and ensures that we aggressively test each Bevy release's migration path.
+- The Bevy editor must be a Bevy application.
+  - This is essential both for dogfooding and to ease development: Bevy contributors all know how to write Bevy!
+
+## Should
```

Comment:
Maybe add that it should be in harmony with / not contradict the CLI? Would be weird if the CLI had a different set of templates than the editor, for example.

Reply:
I think it should probably call into a library shared by the CLI. That's too nitty-gritty for here though: I'd like to make a roadmap stage for CLI + Editor unification down the line as things develop.

Reply:
Agreed. I meant more as a "the editor prototype should not actively make it harder to unify them down the line". In case an eager contributor decides to implement something without checking first whether the CLI already supports it.